lib/internal/errors.js

```diff
@@ -1083,6 +1083,15 @@ E('ERR_HTTP_SOCKET_ENCODING',
 E('ERR_HTTP_TRAILER_INVALID',
   'Trailers are invalid with this transfer encoding', Error);
 E('ERR_ILLEGAL_CONSTRUCTOR', 'Illegal constructor', TypeError);
+E('ERR_IMPORT_ASSERTION_TYPE_FAILED', (url, type) =>
+  `Module ${JSONStringify(`${url}`)} is not of type "${type}"`,
+  TypeError);
+E('ERR_IMPORT_ASSERTION_TYPE_MISSING', (url, expected) =>
+  `Module ${JSONStringify(`${url}`)} needs an import assertion of type "${expected}"`,
+  TypeError);
+E('ERR_IMPORT_ASSERTION_TYPE_UNSUPPORTED', (url, type, expected = null) =>
+  `Invalid import assertion type ${JSONStringify(type)}. ${JSONStringify(`${url}`)} needs ${expected == null ? 'no' : 'an'} import assertion type${expected == null ? '' : ` "${expected}"`}`,
```

Comment:
can this be triggered with `import(url, { type: notAString })`? should there be a different, simpler error message for when the `typeof` the type is not `string`?

Reply:
This throws `TypeError: Import assertion value must be a string` at the V8 level.

Reply:
ah ok, so then this message wouldn't be used for that case? awesome.